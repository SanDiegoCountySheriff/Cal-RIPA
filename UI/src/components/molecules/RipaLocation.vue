<template>
  <div class="ripa-location tw-pb-4">
    <ripa-form-header
      title="Location"
      required
      subtitle="ยง999.226(a)(3)"
      :on-open-statute="onOpenStatute"
    >
    </ripa-form-header>

    <v-container>
      <v-row no-gutters>
        <v-col cols="12" sm="12" md="3" class="tw-pr-2">
          <template v-if="isGeolocationAvailable">
            <div class="tw-mr-2 tw-mt-0 sm:tw-mt-4">
              <v-btn
                class="tw-w-full"
                outlined
                small
                :loading="loadingGps"
                @click="handleCurrentLocation"
              >
                Current Location
              </v-btn>
            </div>
          </template>
        </v-col>
        <v-col cols="12" sm="12" md="3" class="tw-pr-2">
          <div class="tw-mr-2 tw-mt-0 sm:tw-mt-4">
            <v-btn
              class="tw-w-full"
              outlined
              small
              :disabled="!validLastLocation"
              @click="handleLastLocation"
              >Last Location
            </v-btn>
          </div>
        </v-col>
        <v-col cols="12" sm="12" md="3" class="tw-pr-2">
          <div class="tw-mr-2 tw-mt-0 sm:tw-mt-4">
            <v-btn
              class="tw-w-full"
              outlined
              small
              @click="handleOpenFavorites"
            >
              Open Favorites
            </v-btn>
          </div>
        </v-col>
        <v-col cols="12" sm="12" md="3">
          <div class="tw-mr-2 tw-mt-0 sm:tw-mt-4">
            <v-btn
              class="tw-w-full"
              outlined
              small
              @click="handleSaveFavorite"
              :disabled="!isOnlineAndAuthenticated"
            >
              Save Location
            </v-btn>
          </div>
        </v-col>
      </v-row>

      <v-row no-gutters>
        <v-col cols="12" sm="12">
          <ripa-switch
            v-model="model.location.isSchool"
            label="K-12 Public School"
            :max-width="200"
            @input="handleInput"
          ></ripa-switch>

          <template v-if="model.location.isSchool">
            <ripa-autocomplete
              v-model="model.location.school"
              hint="Select 1 School (required)"
              item-text="fullName"
              item-value="cdsCode"
              label="School"
              :items="schools"
              :rules="schoolRules"
              @input="handleInput"
            ></ripa-autocomplete>
          </template>

          <ripa-alert class="tw-mt-8" alert-outlined alert-type="info">
            Note: Do not provide a street address if the location is a
            residence.
          </ripa-alert>
        </v-col>
      </v-row>

      <v-row no-gutters>
        <v-col cols="12" sm="12">
          <div class="md:tw-mr-4">
            <template v-if="model.location.piiFound">
              <ripa-alert alert-outlined alert-type="warning">
                The address may contain personally identifying information.
                Please remove if possible.
              </ripa-alert>
            </template>
          </div>
        </v-col>

        <v-col cols="12" sm="12" md="6">
          <div class="md:tw-mr-4">
            <ripa-number-input
              v-model="model.location.blockNumber"
              label="Block Number"
              :loading="loadingPii"
              :rules="blockNumberRules"
              @input="handleInput"
            >
            </ripa-number-input>
          </div>
        </v-col>

        <v-col cols="12" sm="12" md="6">
          <div>
            <ripa-text-input
              v-model="model.location.streetName"
              label="Street Name"
              :loading="loadingPii"
              :rules="streetNameRules"
              @input="handleInput"
            >
            </ripa-text-input>
          </div>
        </v-col>
      </v-row>

      <v-row no-gutters>
        <v-col cols="12" sm="12">
          <ripa-subheader text="-- or --"></ripa-subheader>

          <ripa-text-input
            v-model="model.location.intersection"
            label="Closest Intersection"
            :loading="loadingPii"
            :rules="intersectionRules"
            @input="handleInput"
          >
          </ripa-text-input>

          <ripa-switch
            v-model="model.location.toggleLocationOptions"
            label="More Location Options"
            :max-width="225"
            @input="handleInput"
          ></ripa-switch>

          <template v-if="model.location.toggleLocationOptions">
            <ripa-subheader text="-- or --"></ripa-subheader>

            <ripa-text-input
              v-model="model.location.highwayExit"
              label="Highway and closest exit"
              :loading="loadingPii"
              :rules="highwayRules"
              @input="handleInput"
            >
            </ripa-text-input>

            <ripa-subheader text="-- or --"></ripa-subheader>

            <ripa-text-input
              v-model="model.location.landmark"
              label="Road marker, landmark, or other"
              :rules="landmarkRules"
              @input="handleInput"
            >
            </ripa-text-input>
          </template>

          <div class="tw-mt-8">
            <ripa-switch
              v-model="model.location.outOfCounty"
              label="City Out of County?"
              :max-width="200"
              @input="handleInput"
            ></ripa-switch>
          </div>
        </v-col>
      </v-row>

      <v-row no-gutters>
        <v-col cols="12" sm="12" md="6">
          <div class="md:tw-mr-4">
            <ripa-autocomplete
              v-model="model.location.city"
              hint="Select 1 City (required)"
              persistent-hint
              item-text="fullName"
              item-value="id"
              label="City"
              :items="getCities"
              :rules="cityRules"
              @input="handleInput"
            ></ripa-autocomplete>
          </div>
        </v-col>

        <template v-if="displayBeatInput">
          <v-col cols="12" sm="12" md="6">
            <div>
              <ripa-autocomplete
                v-model="model.location.beat"
                hint="Select 1 Beat (required)"
                persistent-hint
                item-text="fullName"
                item-value="id"
                label="Beat"
                :items="beats"
                :disabled="model.location.outOfCounty"
                @input="handleInput"
              ></ripa-autocomplete>
            </div>
          </v-col>
        </template>
      </v-row>
    </v-container>
  </div>
</template>

<script>
import RipaAlert from '@/components/atoms/RipaAlert'
import RipaAutocomplete from '@/components/atoms/RipaAutocomplete'
import RipaFormHeader from '@/components/molecules/RipaFormHeader'
import RipaModelMixin from '@/components/mixins/RipaModelMixin'
import RipaNumberInput from '@/components/atoms/RipaNumberInput'
import RipaSubheader from '@/components/atoms/RipaSubheader'
import RipaSwitch from '@/components/atoms/RipaSwitch'
import RipaTextInput from '@/components/atoms/RipaTextInput'

export default {
  name: 'ripa-location',

  mixins: [RipaModelMixin],

  components: {
    RipaAlert,
    RipaAutocomplete,
    RipaFormHeader,
    RipaNumberInput,
    RipaSubheader,
    RipaSwitch,
    RipaTextInput,
  },

  data() {
    return {
      viewModel: this.updateModel(this.value),
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
    },

    isGeolocationAvailable() {
      return navigator.geolocation
    },

    getCities() {
      const checked = this.viewModel.location.outOfCounty

      return checked ? this.nonCountyCities : this.countyCities
    },

    schoolRules() {
      const checked = this.viewModel.location.isSchool
      const school = this.viewModel.location.school

      return [(checked && school !== null) || 'A school is required']
    },

    cityRules() {
      const city = this.viewModel.location.city

      return [city !== null || 'A city is required']
    },

    blockNumberRules() {
      const blockNumber = this.viewModel.location.blockNumber

      return [
        this.isLocationOptionsFilled ||
          blockNumber !== null ||
          'A block number is required',
      ]
    },

    streetNameRules() {
      const streetName = this.viewModel.location.streetName

      return [
        this.isLocationOptionsFilled ||
          (streetName && streetName.length > 0) ||
          'A street name is required',
      ]
    },

    intersectionRules() {
      const intersection = this.viewModel.location.intersection

      return [
        this.isLocationOptionsFilled ||
          (intersection && intersection.length > 0) ||
          'An intersection is required',
      ]
    },

    highwayRules() {
      const checked = this.viewModel.location.toggleLocationOptions
      const highwayExit = this.viewModel.location.highwayExit

      return [
        this.isLocationOptionsFilled ||
          (checked && highwayExit && highwayExit.length > 0) ||
          'A highway and closest exit is required',
      ]
    },

    landmarkRules() {
      const checked = this.viewModel.location.toggleLocationOptions
      const landmark = this.viewModel.location.landmark

      return [
        this.isLocationOptionsFilled ||
          (checked && landmark && landmark.length > 0) ||
          'A road marker, landmark, or other description is required',
      ]
    },

    isLocationOptionsFilled() {
      const blockNumber = this.viewModel.location.blockNumber
      const streetName = this.viewModel.location.streetName
      const intersection = this.viewModel.location.intersection
      const checked = this.viewModel.location.toggleLocationOptions
      const highwayExit = this.viewModel.location.highwayExit
      const landmark = this.viewModel.location.landmark

      const isValid =
        (blockNumber !== null && streetName && streetName.length > 0) ||
        (intersection && intersection.length > 0) ||
        (checked && highwayExit && highwayExit.length > 0) ||
        (checked && landmark && landmark.length > 0)

      return isValid
    },
  },

  methods: {
    handleCurrentLocation() {
      if (navigator.geolocation) {
        if (this.onGpsLocation) {
          this.onGpsLocation()
        }
      } else {
        console.log('Geolocation is not supported by this browser.')
      }
    },

    handleInput() {
      this.updateSchoolModel()
      this.updateBlockNumberModel()
      this.updateFullAddressModel()
      this.updateStopReasonModel()
      this.updateStopResultModel()
      this.$emit('input', this.viewModel)
    },

    handleInputOutOfCounty(newVal) {
      const currentVal = this.viewModel.location.outOfCounty
      if (newVal !== currentVal) {
        this.updateOutOfCountyModel()
        this.handleInput()
      }
    },

    handleLastLocation() {
      if (this.onOpenLastLocation) {
        this.onOpenLastLocation()
      }
    },

    handleOpenFavorites() {
      if (this.onOpenFavorites) {
        this.onOpenFavorites()
      }
    },

    handleSaveFavorite() {
      if (this.onSaveFavorite) {
        this.onSaveFavorite(this.viewModel.location)
      }
    },
  },

  watch: {
    value(newVal) {
      this.viewModel = this.updateModel(newVal)
    },

    lastLocation(newVal) {
      if (newVal) {
        this.viewModel.location = newVal
        this.handleInput()
      }
    },

    'viewModel.location.outOfCounty': {
      handler(newVal, oldVal) {
        if (oldVal !== newVal) {
          this.updateOutOfCountyModel()
        }
      },
    },

    'viewModel.location.blockNumber': {
      handler(newVal, oldVal) {
        if (oldVal !== newVal) {
          this.updateBlockNumberModel()
        }
      },
    },
  },

  props: {
    value: {
      type: Object,
      default: () => {},
    },
    schools: {
      type: Array,
      default: () => [],
    },
    beats: {
      type: Array,
      default: () => [],
    },
    countyCities: {
      type: Array,
      default: () => [],
    },
    displayBeatInput: {
      type: Boolean,
      default: false,
    },
    isOnlineAndAuthenticated: {
      type: Boolean,
      default: false,
    },
    lastLocation: {
      type: Object,
      default: () => {},
    },
    loadingGps: {
      type: Boolean,
      default: false,
    },
    loadingPii: {
      type: Boolean,
      default: false,
    },
    nonCountyCities: {
      type: Array,
      default: () => [],
    },
    validLastLocation: {
      type: Boolean,
      default: false,
    },
    onOpenFavorites: {
      type: Function,
      default: () => {},
    },
    onOpenLastLocation: {
      type: Function,
      default: () => {},
    },
    onSaveFavorite: {
      type: Function,
      default: () => {},
    },
    onGpsLocation: {
      type: Function,
      default: () => {},
    },
  },
}
</script>
