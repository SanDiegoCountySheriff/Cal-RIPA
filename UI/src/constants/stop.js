export const SUBMISSION_STATUSES = [
  {
    text: 'Failed',
    value: 'Failed',
    editable: true,
  },
  {
    text: 'Submitted',
    value: 'Submitted',
    editable: false,
  },
  {
    text: 'Unsubmitted',
    value: 'Unsubmitted',
    editable: true,
  },
  {
    text: 'Resubmitted',
    value: 'Resubmitted',
    editable: false,
  },
]
