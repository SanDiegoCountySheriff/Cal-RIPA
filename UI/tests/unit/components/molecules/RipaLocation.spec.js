import RipaLocation from '@/components/molecules/RipaLocation.vue'
import { shallowMount, mount } from '@vue/test-utils'
import { defaultStop } from '@/utilities/stop.js'
import Vuetify from 'vuetify'
import { computed } from 'vue'

describe('Ripa Location', () => {
  let vuetify
  let wrapper
  let stop

  const mockGeolocation = {
    getCurrentPosition: jest.fn().mockReturnValue(200),
    watchPosition: jest.fn(),
  }

  global.navigator.geolocation = mockGeolocation

  const schoolsList = JSON.parse('[{"cdsCode":"1","fullName":"High School"}]')
  const countyCitiesList = JSON.parse('[{"id":"1","fullName":"CountyCity"}]')
  const nonCountyCitiesList = JSON.parse(
    '[{"id":"2","fullName":"NonCountyCity"}]',
  )
  const beatsList = JSON.parse('[{"id":"1","fullName":"Beat 1"}]')

  beforeEach(() => {
    vuetify = new Vuetify()
    stop = defaultStop()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return mount(RipaLocation, {
      vuetify,
      propsData: {
        ...propsData,
        onOpenFavorites: jest.fn(),
        onOpenLastLocation: jest.fn(),
        onSaveFavorite: jest.fn(),
        onGpsLocation: jest.fn(),
      },
      provide: {
        loadingPiiStep1: computed(() => false),
        loadingGps: computed(() => false),
        isOnlineAndAuthenticated() {
          return true
        },
        beats: computed(() => beatsList),
        countyCities: computed(() => countyCitiesList),
        nonCountyCities: computed(() => nonCountyCitiesList),
        schools: computed(() => schoolsList),
        displayBeatInput() {
          return true
        },
        lastLocation() {
          return {}
        },
        validLastLocation: computed(() => true),
      },
    })
  }

  const schoolTestCases = [
    {
      school: null,
      expected: ['A school is required'],
    },
    {
      school: '1',
      expected: [true],
    },
  ]

  it('should match snapshot', () => {
    wrapper = mount(RipaLocation, {
      vuetify,
      propsData: {
        value: stop,
        onOpenFavorites: jest.fn(),
        onOpenLastLocation: jest.fn(),
        onSaveFavorite: jest.fn(),
        onGpsLocation: jest.fn(),
      },
      provide: {
        loadingPiiStep1: computed(() => false),
        loadingGps: computed(() => false),
        isOnlineAndAuthenticated() {
          return true
        },
        beats: computed(() => beatsList),
        countyCities: computed(() => countyCitiesList),
        nonCountyCities: computed(() => nonCountyCitiesList),
        schools: computed(() => schoolsList),
        displayBeatInput() {
          return true
        },
        lastLocation() {
          return {}
        },
        validLastLocation: computed(() => true),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  schoolTestCases.forEach(test => {
    it(`should validate school: ${test.school} as: ${test.expected}`, () => {
      wrapper = factory({
        value: stop,
        schools: schoolsList,
      })

      stop.location.isSchool = true
      stop.location.school = test.school
      wrapper.vm.model = stop

      expect(wrapper.vm.schoolRules).toEqual(test.expected)
    })
  })

  it('should validate block number', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.blockNumberRules).toEqual([
      'A block number is required',
      'Block number plus street name must be between 5 and 250 characters',
    ])

    stop.location.blockNumber = '1000'
    wrapper.vm.model = stop

    expect(wrapper.vm.blockNumberRules).toStrictEqual([true, true])
  })

  it('should validate street name', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.streetNameRules).toEqual([
      'A street name is required',
      'Block number plus street name must be between 5 and 250 characters',
    ])

    stop.location.streetName = 'Anystreet St'
    wrapper.vm.model = stop

    expect(wrapper.vm.streetNameRules).toStrictEqual([true, true])
  })

  it('should validate intersection', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.vm.intersectionRules).toEqual([
      'An intersection is required',
      'Intersection must be between 5 and 250 characters',
    ])

    stop.location.intersection = ''
    wrapper.vm.model = stop

    expect(wrapper.vm.intersectionRules).toEqual([
      'An intersection is required',
      'Intersection must be between 5 and 250 characters',
    ])

    stop.location.intersection = '5th and Main'
    wrapper.vm.model = stop

    expect(wrapper.vm.streetNameRules).toStrictEqual([true, true])

    stop.location.intersection = null
    stop.location.blockNumber = '1000'
    stop.location.streetName = 'Anystreet St'
    wrapper.vm.model = stop

    expect(wrapper.vm.streetNameRules).toStrictEqual([true, true])

    stop.location.intersection = '5th and Main'
    wrapper.vm.model = stop

    expect(wrapper.vm.streetNameRules).toStrictEqual([true, true])
  })

  it('should validate highway', () => {
    wrapper = factory({ value: stop })
    stop.location.toggleLocationOptions = true
    expect(wrapper.vm.highwayRules).toEqual([
      'A highway and closest exit is required',
      'Highway and closest exit must be between 5 and 250 characters',
    ])

    stop.location.highwayExit = 'Exit 1A'
    wrapper.vm.model = stop

    expect(wrapper.vm.highwayRules).toStrictEqual([true, true])
  })

  it('should validate landmark', () => {
    wrapper = factory({ value: stop })
    stop.location.toggleLocationOptions = true
    expect(wrapper.vm.landmarkRules).toEqual([
      'A road marker, landmark, or other description is required',
      'Road marker, landmark or other description must be between 5 and 250 characters',
    ])

    stop.location.landmark = 'Exit 1A'
    wrapper.vm.model = stop

    expect(wrapper.vm.landmarkRules).toStrictEqual([true, true])
  })

  it('should validate latitude coordinate for v2 stop', () => {
    wrapper = factory({ value: stop })
    stop.location.toggleLocationOptions = true
    expect(wrapper.vm.latitudeRules).toEqual([
      'A valid latitude with a maximum of 3 digits after the decimal is required',
    ])

    stop.location.latitude = '-11.230'
    wrapper.vm.model = stop

    expect(wrapper.vm.latitudeRules).toStrictEqual([true])
  })

  it('should validate longitude coordinate for v2 stop', () => {
    wrapper = factory({ value: stop })
    stop.location.toggleLocationOptions = true
    expect(wrapper.vm.longitudeRules).toEqual([
      'A valid longitude with a maximum of 3 digits after the decimal is required',
    ])

    stop.location.longitude = '-11.230'
    wrapper.vm.model = stop

    expect(wrapper.vm.longitudeRules).toStrictEqual([true])
  })

  it('should return the geolocation score', () => {
    wrapper = factory({ value: stop })
    const getCurrPos = jest.spyOn(
      global.navigator.geolocation,
      'getCurrentPosition',
    )
    wrapper.vm.isGeolocationAvailable()

    expect(getCurrPos).toHaveBeenCalled()
  })

  it.todo('should handle input out of county')
  it.todo('should handle out of county toggle')
})
