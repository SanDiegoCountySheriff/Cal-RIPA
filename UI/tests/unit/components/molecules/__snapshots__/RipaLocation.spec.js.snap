// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Location should match snapshot 1`] = `
<div class="ripa-location tw-pb-4" onopenfavorites="function () {
        return fn.apply(this, arguments);
      }" onopenlastlocation="function () {
        return fn.apply(this, arguments);
      }" onsavefavorite="function () {
        return fn.apply(this, arguments);
      }" ongpslocation="function () {
        return fn.apply(this, arguments);
      }">
  <div class="ripa-form-header">
    <div class="row tw-ml-1 tw-mt-1"><span class="v-tooltip v-tooltip--top"><!----><i aria-hidden="true" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate mdi mdi-alert theme--light error--text" style="font-size: 16px;"></i></span> <button type="button" class="v-icon notranslate icon v-icon--link mdi mdi-scale-balance theme--light primary--text" style="font-size: 16px;"></button>
      <div class="ripa-header"><span class="tw-text-base md:tw-text-lg tw-uppercase">Location</span></div>
    </div>
  </div>
  <div class="container mt-5">
    <div class="row">
      <div class="col-sm-6 col-md-3 col-12"><button type="button" class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary"><span class="v-btn__content">
          Current Location
        </span></button></div>
      <div class="col-sm-6 col-md-3 col-12"><button type="button" class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary"><span class="v-btn__content">
          Last Location
        </span></button></div>
      <div class="col-sm-6 col-md-3 col-12"><button type="button" class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary"><span class="v-btn__content">
          Open Favorites
        </span></button></div>
      <div class="col-sm-6 col-md-3 col-12"><button type="button" class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg theme--light v-size--small primary"><span class="v-btn__content">
            Save Location
          </span></button></div>
    </div>
    <div class="row">
      <!---->
    </div>
    <div class="row">
      <div class="text-center col"></div>
    </div>
    <div class="row">
      <div class="col-sm-12 col-12">
        <div class="ripa-switch" style="max-width: 200px;">
          <div class="v-input v-input--reverse v-input--expand v-input--hide-details theme--light v-input--selection-controls v-input--switch v-input--switch--inset">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-12" role="switch" type="checkbox" aria-disabled="false" value="">
                  <div class="v-input--selection-controls__ripple"></div>
                  <div class="v-input--switch__track theme--light"></div>
                  <div class="v-input--switch__thumb theme--light">
                    <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                  </div>
                </div><label for="input-12" class="v-label theme--light" style="left: 0px; position: relative;"><span class="main-label mr-3">
        K-12 Public School
      </span> <span class="switch-label" style="right: -45px;">
        No
      </span></label>
              </div>
            </div>
          </div>
        </div>
        <!---->
        <div role="alert" class="v-alert mt-4 v-sheet v-sheet--outlined theme--light v-alert--dense v-alert--outlined info--text">
          <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-information theme--light info--text"></i>
            <div class="v-alert__content">
              Note: Do not provide a street address if the location is a
              residence.
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
    </div>
    <div class="row">
      <div class="col-sm-12 col-md-6 col-12">
        <div class="ripa-text-input">
          <div class="v-input theme--light v-text-field">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-text-field__slot"><label for="input-18" class="v-label theme--light" style="left: 0px; position: absolute;">Block Number</label><input id="input-18" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <!---->
        </div>
      </div>
      <div class="col-sm-12 col-md-6 col-12">
        <div class="ripa-text-input">
          <!---->
          <div class="v-input theme--light v-text-field">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-text-field__slot"><label for="input-22" class="v-label theme--light" style="left: 0px; position: absolute;">Street Name</label><input id="input-22" type="text"></div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="row">
      <div class="col-sm-12 col-12">
        <div class="ripa-label tw-font-bold tw-my-4"><span>
    -- or --
  </span></div>
        <div class="row">
          <div class="col-sm-6 col-12">
            <div class="ripa-text-input">
              <!---->
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-28" class="v-label theme--light" style="left: 0px; position: absolute;">Cross Street 1</label><input id="input-28" type="text"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div class="col-sm-6 col-12">
            <div class="ripa-text-input">
              <!---->
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-32" class="v-label theme--light" style="left: 0px; position: absolute;">Cross Street 2</label><input id="input-32" type="text"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="ripa-switch" style="max-width: 225px;">
          <div class="v-input v-input--reverse v-input--expand v-input--hide-details theme--light v-input--selection-controls v-input--switch v-input--switch--inset">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-36" role="switch" type="checkbox" aria-disabled="false" value="">
                  <div class="v-input--selection-controls__ripple"></div>
                  <div class="v-input--switch__track theme--light"></div>
                  <div class="v-input--switch__thumb theme--light">
                    <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                  </div>
                </div><label for="input-36" class="v-label theme--light" style="left: 0px; position: relative;"><span class="main-label mr-3">
        More Location Options
      </span> <span class="switch-label" style="right: -45px;">
        No
      </span></label>
              </div>
            </div>
          </div>
        </div>
        <!---->
        <div class="tw-mt-8">
          <div class="ripa-switch" style="max-width: 200px;">
            <div class="v-input v-input--reverse v-input--expand v-input--hide-details theme--light v-input--selection-controls v-input--switch v-input--switch--inset">
              <div class="v-input__control">
                <div class="v-input__slot">
                  <div class="v-input--selection-controls__input"><input aria-checked="false" id="input-39" role="switch" type="checkbox" aria-disabled="false" value="">
                    <div class="v-input--selection-controls__ripple"></div>
                    <div class="v-input--switch__track theme--light"></div>
                    <div class="v-input--switch__thumb theme--light">
                      <transition-stub name="fab-transition" mode="out-in"></transition-stub>
                    </div>
                  </div><label for="input-39" class="v-label theme--light" style="left: 0px; position: relative;"><span class="main-label mr-3">
        City Out of County?
      </span> <span class="switch-label" style="right: -45px;">
        No
      </span></label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="row">
      <div class="col-sm-12 col-md-6 col-12">
        <div class="v-input tw-my-4 theme--light v-text-field v-text-field--solo-flat v-select v-autocomplete">
          <div class="v-input__control">
            <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-42" class="v-input__slot">
              <div class="v-select__slot"><label for="input-42" class="v-label theme--light" style="left: 0px; position: absolute;">City or Unincorporated Area</label><input id="input-42" type="text" autocomplete="off">
                <div class="v-input__append-inner">
                  <div></div>
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper">
                  <div class="v-messages__message">Select 1 City (required)</div>
                </transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-12 col-md-6 col-12">
        <div class="v-input tw-my-4 theme--light v-text-field v-text-field--solo-flat v-select v-autocomplete">
          <div class="v-input__control">
            <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-48" class="v-input__slot">
              <div class="v-select__slot"><label for="input-48" class="v-label theme--light" style="left: 0px; position: absolute;">Beat</label><input id="input-48" type="text" autocomplete="off">
                <div class="v-input__append-inner">
                  <div></div>
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper">
                  <div class="v-messages__message">Select 1 Beat (required)</div>
                </transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
